Userland/Libraries/LibWeb/DOM/ParentNode.cpp

```diff
@@ -13,17 +13,22 @@ namespace Web::DOM {
 
 RefPtr<Element> ParentNode::query_selector(const StringView& selector_text)
 {
-    auto selector = parse_selector(CSS::DeprecatedParsingContext(*this), selector_text);
-    if (!selector)
+    auto maybe_selectors = parse_selector(CSS::ParsingContext(*this), selector_text);
```

Comment:
Why did this suddenly go from one selector to many?

Reply:
This is described in the `LibWeb: Implement CSS parsing convenience functions` commit right before where the method is added:
> These mostly match the API in `DeprecatedCSSParser.h`. The exception is
that `parse_selector()` returns a `SelectorList` instead of just one
`Selector`. The only uses of that are in
`ParentNode::query_selector[_all]()` which should be matching against a
list, according to the spec.

Reply:
Maybe the function should be renamed `parse_selector_list()` for clarity. The outside world shouldn't have to deal with the "17 different things are called a selector" madness.

Reply:
I was more concerned with the semantics of this API, as in,, whether it even supports multiple selectors, and if passing multiple selectors means "matches any one of these" or "matches all of these"

Reply:
Good catch, I made some assumptions I shouldn't have.

I've now checked, and [the querySelector/querySelectorAll spec](https://dom.spec.whatwg.org/#ref-for-dom-parentnode-queryselector%E2%91%A0) specifies (a couple of levels down) using the [match a selector against a tree](https://www.w3.org/TR/selectors-4/#match-against-tree) algorithm, which uses the [match a selector against an element](https://www.w3.org/TR/selectors-4/#match-against-element) algorithm, which treats the "selector" as a list of complex-selectors (called `Selector` in our code), and finds elements that match **any** of the complex-selectors in the list. So, the way I wrote it is correct I believe, but I should be more careful in future.

Reply:
I think the only thing left to do then, is to mention this bit in the commit message, and then we can :merge:

Reply:
:+1: